src/main/java/org/takes/rs/RsWithHeader.java

```diff
@@ -94,22 +95,61 @@ public RsWithHeader(final Response res, final CharSequence name,
      */
     public RsWithHeader(final Response res, final CharSequence header) {
         super(
-            new Response() {
-                @Override
-                public List<String> head() throws IOException {
-                    final List<String> head = new LinkedList<String>();
-                    for (final String hdr : res.head()) {
-                        head.add(hdr);
-                    }
-                    head.add(header.toString());
-                    return head;
-                }
-                @Override
-                public InputStream body() throws IOException {
-                    return res.body();
-                }
-            }
+            new RsWithHeader.ResponseImpl(res, header)
+        );
+    }
+
+    private static final class ResponseImpl implements Response {
+        /**
+         * Pattern for all other lines in the head.
+         */
+        private static final Pattern HEADER = Pattern.compile(
+            "[a-zA-Z\\-]+:\\p{Print}+"
         );
+
+        /**
+         * Original response.
+         */
+        private final transient Response response;
+
+        /**
+         * Header to add.
+         */
+        private final transient CharSequence value;
+
+        /**
+         * Ctor.
+         * @param res Original response
+         * @param header Header to add
+         */
+        ResponseImpl(final Response res, final CharSequence header) {
+            if (!HEADER.matcher(header).matches()) {
+                throw new IllegalArgumentException(
```

Comment:
@alevohin please look at previous version.Throwing exception here - result of passing incorrect values (not general, problem, its will be thrown only if programmer write incorrect code, and use incorrect value). Validation can be done only in constructor, otherwise it will be thrown in some anther place, after creating instance class with wrong header. We need be sure, that object is valid. "Without exceptions, there should be only one constructor that stores data in object variables" - as I understand @yegor256 mean here about "one, only one constructor...", nothing about throw exceptions.  I will change implementation, and exception will be thrown in method head, what you think about it (like here org.takes.rs.RsPrint#printHead(java.io.OutputStream))?